application/common/models/User.php

```diff
@@ -426,13 +435,42 @@ public function getDisplayName()
     }
 
     /**
-     * Is it time to nag the user to add an MFA?
-     *
-     * @return bool
+     * Based on current time and presence of MFA and Method options,
+     * determine which "nag" to present to the user.
      */
-    public function isTimeToNagForMfa()
+    public function getNagState()
     {
-        return strtotime($this->nag_for_mfa_after) < time();
+        /*
+         * Don't recalculate in case the date has changed since the last calculation.
+         */
+        if ($this->nagState !== null) return $this->nagState;
```

Comment:
Per PSR-2, you'll want to switch this to enclose the if-block in curly braces:
```suggestion
        if ($this->nagState !== null) {
            return $this->nagState;
        }
```